contract/src/main.rs

```diff
@@ -1386,56 +1386,54 @@ pub extern "C" fn updated_receipts() {
 
 #[no_mangle]
 pub extern "C" fn register_owner() {
-    if let OwnerReverseLookupMode::NoLookUp = utils::get_reporting_mode() {
```

Comment:
Before you were treating the exception NoLookUp and returning, now you surround with a big if with Complete , and it only affect burn.rs tests ? I would expect a change on that statement to trigger a test on InvalidReportingMode being returned but maybe it was not covered  or there is no point covering that.